sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/resourcehints/ResourceHints.java

```diff
@@ -207,8 +210,25 @@ public int hashCode() {
     }
   }
 
-  /** Sets desired minimal available RAM size to have in transform's execution environment. */
+  /**
+   * Sets desired minimal available RAM size to have in transform's execution environment.
+   *
+   * @param ramBytes specifies a positive RAM size in bytes. A number greater than 1G
+   *     (1_000_000_000L) is typical.
+   */
   public ResourceHints withMinRam(long ramBytes) {
```

Comment:
What if we took a Long rather than a long? In this case, ints would not be accidentally promoted. (We'd want to drop a comment as to why.)

Reply:
Warned user for changing function signature as part of TODO